lxml-stubs/etree.pyi

```diff
@@ -470,3 +471,26 @@ class XPath:
         _etree_or_element: Union[_Element, _ElementTree],
         **_variables: _XPathObject
     ) -> _XPathObject: ...
+
+_ElementFactory = Callable[[Any, Dict[Any, Any]], _Element]
+_CommentFactory = Callable[[Optional[_AnyStr]], _Comment]
+_ProcessingInstructionFactory = Callable[
+    [_AnyStr, Optional[_AnyStr]], _ProcessingInstruction
+]
+
+class TreeBuilder:
+    def __init__(
+        self,
+        element_factory: Optional[_ElementFactory] = ...,
+        parser: Optional[_BaseParser] = ...,
+        comment_factory: Optional[_CommentFactory] = ...,
+        pi_factory: Optional[_ProcessingInstructionFactory] = ...,
+        insert_comments: bool = ...,
+        insert_pis: bool = ...,
+    ) -> None: ...
+    def close(self) -> _Element: ...
+    def comment(self, text: _AnyStr) -> None: ...
+    def data(self, data: _AnyStr) -> None: ...
+    def end(self, tag: _AnyStr) -> None: ...
+    def pi(self, target: _AnyStr, data: Optional[_AnyStr] = ...) -> Any: ...
+    def start(self, tag: _AnyStr, attrib: _DictAnyStr) -> None: ...
```

Comment:
I don't think `_DictAnyStr` is correct here (and probably elsewhere). There is no guarantee that a unicode name will map to a unicode value and vice versa (in Py2, their type is independent). In Py3, it's always unicode text anyway.

Reply:
@scoder I agree, thought about that too - however, I have decided not to go against the current typing of the `ParserTarget` protocol. Now that you have mentioned it, I have adjusted the typing of `ParserTarget.start` as well.

Reply:
I've also added a test that should fail if `TreeBuilder` becomes incompatbile to the `ParserTarget` spec.